src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -47,46 +43,22 @@ public MemoryCache(IOptions<MemoryCacheOptions> optionsAccessor)
         /// <param name="loggerFactory">The factory used to create loggers.</param>
         public MemoryCache(IOptions<MemoryCacheOptions> optionsAccessor, ILoggerFactory loggerFactory)
         {
-            if (optionsAccessor == null)
-            {
-                throw new ArgumentNullException(nameof(optionsAccessor));
-            }
-
-            if (loggerFactory == null)
-            {
-                throw new ArgumentNullException(nameof(loggerFactory));
-            }
-
-            _options = optionsAccessor.Value;
-            _logger = loggerFactory.CreateLogger<MemoryCache>();
+            _options = optionsAccessor?.Value ?? throw new ArgumentNullException(nameof(optionsAccessor));
+            _logger = loggerFactory?.CreateLogger<MemoryCache>() ?? throw new ArgumentNullException(nameof(loggerFactory));
```

Comment:
If ` loggerFactory.CreateLogger` returns `null`, is it appropriate to throw an `ArgumentNullException` exception? `loggerFactory` wasn't null - what it returned was.